src/Abc.Zebus/Directory/PeerDirectoryClient.cs

```diff
@@ -27,13 +31,15 @@ public partial class PeerDirectoryClient : IPeerDirectory,
         private readonly UniqueTimestampProvider _timestampProvider = new UniqueTimestampProvider(10);
         private readonly IBusConfiguration _configuration;
         private readonly Stopwatch _pingStopwatch = new Stopwatch();
+        private readonly IMessageDispatcher _messageDispatcher;
         private BlockingCollection<IEvent> _messagesReceivedDuringRegister;
         private IEnumerable<Peer> _directoryPeers;
         private Peer _self;
 
-        public PeerDirectoryClient(IBusConfiguration configuration)
+        public PeerDirectoryClient(IBusConfiguration configuration, IMessageDispatcher messageDispatcher)
```

Comment:
Not sure I like this dependency, it feels off. Wouldn't it be better if the bus forwarded this info from the peer directory through an event?

Reply:
The `PeerDirectoryClient` also needs to know the subscribed message handlers in order to determine what should be raised through an event so I'm not sure how this could be implmeneted without injecting the `MessageDispatcher`.  I'm open to any suggestions though.

Reply:
Maybe this should be the responsibility of the bus itself?

Reply:
What do you think @ocoanet?